[0.125s] Invoking command in '/home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing': AMENT_PREFIX_PATH=/home/phr/vscode_project/ros2_launch_testing_ws/install/my_testing:/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms:/home/phr/vscode_project/ros2_launch_testing_ws/install/my_interfaces:${AMENT_PREFIX_PATH} CMAKE_PREFIX_PATH=/home/phr/vscode_project/ros2_launch_testing_ws/install/my_interfaces LD_LIBRARY_PATH=/home/phr/vscode_project/ros2_launch_testing_ws/install/my_interfaces/lib:${LD_LIBRARY_PATH} PYTEST_ADDOPTS=--tb=short --junit-xml=/home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/pytest.xml --junit-prefix=my_testing -o cache_dir=/home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/.pytest_cache -s PYTHONDONTWRITEBYTECODE=1 PYTHONPATH=/home/phr/vscode_project/ros2_launch_testing_ws/install/my_testing/lib/python3.10/site-packages:/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/python3.10/site-packages:/home/phr/vscode_project/ros2_launch_testing_ws/install/my_interfaces/local/lib/python3.10/dist-packages:${PYTHONPATH} /usr/bin/python3 -m pytest
[0.260s] [1m============================= test session starts ==============================[0m
[0.260s] platform linux -- Python 3.10.12, pytest-6.2.5, py-1.10.0, pluggy-0.13.0
[0.262s] cachedir: /home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/.pytest_cache
[0.262s] rootdir: /home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing
[0.262s] plugins: launch-testing-1.0.10, ament-lint-0.12.14, ament-copyright-0.12.14, ament-xmllint-0.12.14, launch-testing-ros-0.19.10, ament-pep257-0.12.14, ament-flake8-0.12.14, colcon-core-0.19.0, cov-3.0.0
[0.326s] [1mcollecting ... [0m[1m
[0.326s] collected 1 item                                                               [0m
[0.326s] 
[0.334s] test/test_integration.py [INFO] [launch]: All log files can be found below /home/phr/.ros/log/2025-08-05-23-43-12-494340-PHR-61449
[0.334s] [INFO] [launch]: Default logging verbosity is set to INFO
[0.345s] [INFO] [number_publisher-1]: process started with pid [61463]
[0.346s] [INFO] [number_processor-2]: process started with pid [61465]
[0.370s] test_communication_and_logic (test_integration.TestIntegration) ... [0m[INFO] [1754408592.537423937] [test_subscriber_node]: 测试节点test_subscriber_node已创建[0m
[0.448s] [number_publisher-1] [INFO] [1754408592.615214744] [number_publisher]: 数字发布节点已启动，将从1开始发布...
[0.450s] [number_processor-2] [INFO] [1754408592.617595210] [number_processor]: 数字处理节点已启动，等待接收数字...
[1.442s] [number_publisher-1] [INFO] [1754408593.609744134] [number_publisher]: 发布: 1
[1.442s] [number_processor-2] [INFO] [1754408593.610080130] [number_processor]: 收到: 1
[1.443s] [number_processor-2] [INFO] [1754408593.610498047] [number_processor]: 发布处理后结果: 2
[2.443s] [number_publisher-1] [INFO] [1754408594.609883395] [number_publisher]: 发布: 2
[2.443s] [number_processor-2] [INFO] [1754408594.610173087] [number_processor]: 收到: 2
[2.443s] [number_processor-2] [INFO] [1754408594.610513832] [number_processor]: 发布处理后结果: 4
[3.442s] [number_publisher-1] [INFO] [1754408595.609690263] [number_publisher]: 发布: 3
[3.442s] [number_processor-2] [INFO] [1754408595.610061877] [number_processor]: 收到: 3
[3.443s] [number_processor-2] [INFO] [1754408595.610368191] [number_processor]: 发布处理后结果: 6
[3.443s] [0m[INFO] [1754408595.610914333] [test_subscriber_node]: 检查接收到的消息。 Got: [2, 4, 6][0m
[3.445s] [0m[INFO] [1754408595.613409965] [test_subscriber_node]: 测试结束，销毁测试节点[0m
[3.446s] ok
[3.450s] test_processor_starts_ok (test_integration.TestIntegration) ... [0m[INFO] [1754408595.617738735] [test_subscriber_node]: 测试节点test_subscriber_node已创建[0m
[3.450s] [0m[INFO] [1754408595.617966690] [test_subscriber_node]: 正在检查 processor 节点的启动日志...[0m
[3.450s] [0m[INFO] [1754408595.618198207] [test_subscriber_node]: Processor 节点启动正常。[0m
[3.452s] [0m[INFO] [1754408595.619928052] [test_subscriber_node]: 测试结束，销毁测试节点[0m
[3.452s] ok
[3.456s] test_publisher_starts_ok (test_integration.TestIntegration) ... [0m[INFO] [1754408595.623580081] [test_subscriber_node]: 测试节点test_subscriber_node已创建[0m
[3.456s] [0m[INFO] [1754408595.623796163] [test_subscriber_node]: 正在检查 publisher 节点的启动日志...[0m
[3.456s] [0m[INFO] [1754408595.624006309] [test_subscriber_node]: Publisher 节点启动正常。[0m
[3.458s] [0m[INFO] [1754408595.625700537] [test_subscriber_node]: 测试结束，销毁测试节点[0m
[3.458s] ok
[3.458s] 
[3.458s] ----------------------------------------------------------------------
[3.458s] Ran 3 tests in 3.123s
[3.458s] 
[3.458s] OK
[3.459s] [INFO] [number_processor-2]: sending signal 'SIGINT' to process[number_processor-2]
[3.459s] [INFO] [number_publisher-1]: sending signal 'SIGINT' to process[number_publisher-1]
[3.459s] [number_processor-2] Traceback (most recent call last):
[3.459s] [number_processor-2]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor", line 33, in <module>
[3.459s] [number_processor-2]     sys.exit(load_entry_point('my-algorithms==0.0.0', 'console_scripts', 'number_processor')())
[3.459s] [number_processor-2]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/python3.10/site-packages/my_algorithms/number_processor_node.py", line 32, in main
[3.460s] [number_processor-2]     rclpy.spin(node)
[3.460s] [number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/__init__.py", line 229, in spin
[3.460s] [number_processor-2]     executor.spin_once()
[3.460s] [number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 751, in spin_once
[3.460s] [number_processor-2]     self._spin_once_impl(timeout_sec)
[3.460s] [number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 740, in _spin_once_impl
[3.460s] [number_processor-2]     handler, entity, node = self.wait_for_ready_callbacks(timeout_sec=timeout_sec)
[3.460s] [number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 723, in wait_for_ready_callbacks
[3.460s] [number_processor-2]     return next(self._cb_iter)
[3.460s] [number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 620, in _wait_for_ready_callbacks
[3.460s] [number_processor-2]     wait_set.wait(timeout_nsec)
[3.460s] [number_processor-2] KeyboardInterrupt
[3.460s] [number_publisher-1] Traceback (most recent call last):
[3.460s] [number_publisher-1]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_publisher", line 33, in <module>
[3.460s] [number_publisher-1]     sys.exit(load_entry_point('my-algorithms==0.0.0', 'console_scripts', 'number_publisher')())
[3.460s] [number_publisher-1]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/python3.10/site-packages/my_algorithms/number_publisher_node.py", line 25, in main
[3.460s] [number_publisher-1]     rclpy.spin(node)
[3.460s] [number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/__init__.py", line 229, in spin
[3.461s] [number_publisher-1]     executor.spin_once()
[3.461s] [number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 751, in spin_once
[3.461s] [number_publisher-1]     self._spin_once_impl(timeout_sec)
[3.461s] [number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 740, in _spin_once_impl
[3.461s] [number_publisher-1]     handler, entity, node = self.wait_for_ready_callbacks(timeout_sec=timeout_sec)
[3.461s] [number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 723, in wait_for_ready_callbacks
[3.461s] [number_publisher-1]     return next(self._cb_iter)
[3.461s] [number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 620, in _wait_for_ready_callbacks
[3.461s] [number_publisher-1]     wait_set.wait(timeout_nsec)
[3.461s] [number_publisher-1] KeyboardInterrupt
[3.480s] [ERROR] [number_processor-2]: process has died [pid 61465, exit code -2, cmd '/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor --ros-args -r __node:=number_processor'].
[3.481s] [ERROR] [number_publisher-1]: process has died [pid 61463, exit code -2, cmd '/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_publisher --ros-args -r __node:=number_publisher'].
[3.482s] 
[3.482s] ----------------------------------------------------------------------
[3.482s] Ran 0 tests in 0.000s
[3.482s] 
[3.482s] OK
[3.486s] [32m.[0m
[3.486s] 
[3.486s] - generated xml file: /home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/pytest.xml -
[3.486s] [32m============================== [32m[1m1 passed[0m[32m in 3.23s[0m[32m ===============================[0m
[3.627s] Invoked command in '/home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing' returned '0': AMENT_PREFIX_PATH=/home/phr/vscode_project/ros2_launch_testing_ws/install/my_testing:/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms:/home/phr/vscode_project/ros2_launch_testing_ws/install/my_interfaces:${AMENT_PREFIX_PATH} CMAKE_PREFIX_PATH=/home/phr/vscode_project/ros2_launch_testing_ws/install/my_interfaces LD_LIBRARY_PATH=/home/phr/vscode_project/ros2_launch_testing_ws/install/my_interfaces/lib:${LD_LIBRARY_PATH} PYTEST_ADDOPTS=--tb=short --junit-xml=/home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/pytest.xml --junit-prefix=my_testing -o cache_dir=/home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/.pytest_cache -s PYTHONDONTWRITEBYTECODE=1 PYTHONPATH=/home/phr/vscode_project/ros2_launch_testing_ws/install/my_testing/lib/python3.10/site-packages:/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/python3.10/site-packages:/home/phr/vscode_project/ros2_launch_testing_ws/install/my_interfaces/local/lib/python3.10/dist-packages:${PYTHONPATH} /usr/bin/python3 -m pytest
